### PR TITLE
implement-google-calendar

### PR DESCRIPTION
**Pull request by vldmrklnk**

### CODE REVIEW

File: InternshipBE/BL/Interfaces/IGoogleCalendarService.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Threading.Tasks;
+
+namespace BL.Interfaces
+{
+	public interface IGoogleCalendarService
+	{
+		public Task GetCalendarEvents();
```
Comment (romanstat) [inline InternshipBE/BL/Interfaces/IGoogleCalendarService.cs]: Async

File: InternshipBE/BL/Services/GoogleCalendarService.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+		public GoogleCalendarService(IGoogleConfig googleConfig)
+		{
+			_googleConfig = googleConfig;
+		}
+
+		public Task GetCalendarEvents()
+		{
+			throw new NotImplementedException();
+		}
```
Comment (romanstat) [inline InternshipBE/BL/Services/GoogleCalendarService.cs]: implement

File: InternshipBE/BL/Services/BestContactTimeService.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		{
+			_unitOfWork = unitOfWork;
+			_mapper = mapper;
+			_userManager = userManager;
+		}
+
+		public async Task<List<BestContactTimeDTO>> GetAllBestTimeByIdAsync(string interviewerId)
+		{
+			var timeList = await _unitOfWork.BestContactTime.GetAllByUserIdAsync(interviewerId);
+			var models = _mapper.Map<List<BestContactTimeDTO>>(timeList);
```
Comment (romanstat) [inline InternshipBE/BL/Services/BestContactTimeService.cs]: return _mapper.Map<List<BestContactTimeDTO>>(timeList);

File: InternshipBE/BL/Services/GoogleCalendarService.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+		private readonly IGoogleConfig _googleConfig;
+		private CalendarService _calendarService;
+
+		public GoogleCalendarService(IGoogleConfig googleConfig, IUnitOfWork unitOfWork, UserManager<User> userManager)
+		{
+			_userManager = userManager;
+			_googleConfig = googleConfig;
+		}
+
+		public async Task CreateEventInCalendarAsync(string interviewerID, BestContactTimeDTO model)
```
Comment (romanstat) [inline InternshipBE/BL/Services/GoogleCalendarService.cs]: string interviewerEmail?

File: InternshipBE/DAL/Migrations/20211108145926_sdgfsdgs.cs [hunk 5]
```diff
@@ -3,7 +3,7 @@
 
 namespace DAL.Migrations
 {
-    public partial class InitialDb : Migration
+    public partial class sdgfsdgs : Migration
```
Comment (romanstat) [inline InternshipBE/DAL/Migrations/20211108145926_sdgfsdgs.cs]: naming

File: InternshipBE/DAL/Repositories/BestContactTimeRepository.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+{
+	public class BestContactTimeRepository : GenericRepository<BestContactTime>, IBestContactTimeRepository
+	{
+		public BestContactTimeRepository(ApplicationDbContext context) : base(context)
+		{
+		}
+
+		public async Task<List<BestContactTime>> GetAllByUserIdAsync(string userId)
+		{
+			return await _context.BestContactTimes.Where(x => x.User.Id == userId).ToListAsync();
```
Comment (romanstat) [inline InternshipBE/DAL/Repositories/BestContactTimeRepository.cs]: Include

Comment (romanstat) [reply InternshipBE/DAL/Repositories/BestContactTimeRepository.cs]: you can add UserId filed to BestContactTime
image

File: InternshipBE/WebApi/Controllers/BestContactTimeController.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+			await _bestContactTimeService.SaveBestContactTimeAsync(userName, model);
+
+			return Ok();
+		}
+
+		[HttpGet("GetAllBestContactTimeByUserId")]
+		[Authorize]
+		public async Task<IActionResult> GetAllBestContactTimeByUserId([FromBody] string interviewerId)
+		{
+			var models = await _bestContactTimeService.GetAllBestTimeByIdAsync(interviewerId);
```
Comment (romanstat) [inline InternshipBE/WebApi/Controllers/BestContactTimeController.cs:36]: also can var userName = User.Identity.Name;

File: InternshipBE/WebApi/Controllers/GoogleCalendarController.cs [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+		private readonly IGoogleCalendarService _googleCalendarService;
+
+		public GoogleCalendarController(IGoogleCalendarService googleCalendarService)
+		{
+			_googleCalendarService = googleCalendarService;
+		}
+
+		[HttpPost("setEventToGoogleCalendar")]
+		[Authorize]
+		public async Task<IActionResult> SetEventToGoogleCalendar([FromBody] string interviewerId, BestContactTimeDTO model)
```
Comment (romanstat) [inline InternshipBE/WebApi/Controllers/GoogleCalendarController.cs]: same

File: InternshipBE/WebApi/Controllers/BestContactTimeController.cs [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+			return Ok();
+		}
+
+		[HttpGet("{interviewerId}")]
+		[Authorize]
+		public async Task<IActionResult> GetAllBestContactTimeByUserId(string interviewerId)
+		{
+			var models = await _bestContactTimeService.GetAllBestTimeByIdAsync(interviewerId);
+
+			return Ok(new { models });
```
Comment (romanstat) [inline InternshipBE/WebApi/Controllers/BestContactTimeController.cs]: return Ok(models)?

**Review by romanstat — APPROVED** [review]

**Review by Xodoc — APPROVED** [review]

**Review by Evgeniy-Sinicin — APPROVED** [review]

**Pull request merged**